### PR TITLE
Add Cross section CSG based

### PR DESCRIPTION
**Pull request by ostatni5**
image

### CODE REVIEW

**Comment by grzanka** [comment]
Looks very good ! Are you planning to add anything else to that PR ?

File: src/ThreeEditor/js/viewport/Viewport.ClippedView2.js [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+	this.editor = editor;
+
+	// default order is zero, we assign higher order to stencil plane to display it in front of other objects
+	const STENCIL_RENDER_ORDER = 1;
+
+	// several objects we are using (stencil plan) are limited to a square on a plane with given size (side length)
+	// it limits range of applicability of clipping plane
+	// it also limits the size (horizontally and vertically) of stencil plane
+	// here side length is set to 10 cm
+	const CLIPPING_SIZE = 200;
```
Comment (grzanka) [inline src/ThreeEditor/js/viewport/Viewport.ClippedView2.js:36]: Will this work with a 40x40x40 cm cube ?

Comment (grzanka) [reply src/ThreeEditor/js/viewport/Viewport.ClippedView2.js:36]: What about this project https://gist.githubusercontent.com/grzanka/cd47cacce2f6d5986397630d76580d11/raw/e6ca6a23fbae78b04ddcfaf2582bc7c3f972e995/2022let_plan2.json ?

Comment (ostatni5) [reply src/ThreeEditor/js/viewport/Viewport.ClippedView2.js]: I think yes

**Review by grzanka — APPROVED** [review]
LGTM

File: src/ThreeEditor/js/viewport/Viewport.ClippedViewCSG.ts [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+	name: string,
+	editor: Editor,
+	viewport: typeof Viewport,
+	planeHelpers: THREE.Group,
+	initialObjects: T[],
+	signalGeometryChanged: Signal<T>,
+	signalGeometryAdded: Signal<T>,
+	signalGeometryRemoved: Signal<T>,
+	container: HTMLElement,
+	{ clipPlane, planeHelperColor, planePosLabel }: { clipPlane: THREE.Plane; planeHelperColor: number; planePosLabel: string; }
```
Comment (Derstilon) [inline src/ThreeEditor/js/viewport/Viewport.ClippedViewCSG.ts:34]: Maybe name this object type

File: src/ThreeEditor/js/viewport/Viewport.ClippedViewCSG.ts [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+	const planeHelper = new THREE.PlaneHelper(
+		clipPlane,
+		INTERSECTION_SIZE,
+		planeHelperColor ?? 0xffff00
+	); // default helper color is yellow
+	planeHelpers.add(planeHelper);
+
+	const planePosProperty = `${planePosLabel ?? 'PlanePos'} ${editor.unit.name}`;
+
+	const uiProps: Record<string, any> = {
```
Comment (Derstilon) [inline src/ThreeEditor/js/viewport/Viewport.ClippedViewCSG.ts:60]:
```suggestion
	const uiProps: Record<string, number | boolean> = {
```

File: src/ThreeEditor/js/viewport/Viewport.js [hunk 4]
```diff
@@ -124,26 +124,33 @@ export function Viewport(
 
 		if (!renderer) return;
 
-		if (clipPlane) renderer.clippingPlanes = [clipPlane];
+		// if (clipPlane) renderer.clippingPlanes = [clipPlane];
```
Comment (Derstilon) [inline src/ThreeEditor/js/viewport/Viewport.js:127]:
```suggestion
```

File: src/ThreeEditor/js/viewport/Viewport.js [hunk 5]
```diff
@@ -124,26 +124,33 @@ export function Viewport(
...
+
+			renderer.render(detectManager, camera);
 		}
 
+
+		if (clipPlane)
+			renderer.render(viewClipPlane.scene, camera);
+
+
 		renderer.clippingPlanes = []; // clear clipping planes for next renders
```
Comment (Derstilon) [inline src/ThreeEditor/js/viewport/Viewport.js:154]:
```suggestion
```
I think this is obsolete

**Review by Derstilon — APPROVED** [review]
image

**Review by Derstilon — DISMISSED** [review]

File: src/ThreeEditor/js/viewport/Viewport.js (hunk 4 above)
Comment (ostatni5) [inline src/ThreeEditor/js/viewport/Viewport.js]:
```suggestion
```

**Pull request merged**